### ISSUE TITLE
[Feature Request] How to implement algorithms with multiple optimise phase like PPG?

### ISSUE THREAD

**Issue by levilovearch** [pytorch/rl#977]
## Motivation

I'm trying to implement [PPG](https://proceedings.mlr.press/v139/cobbe21a) and [DNA](https://arxiv.org/pdf/2206.10027.pdf) algorithms with torchrl, and both algorithms have more than one optimise phase in a single training loop. However, I suggest the [Trainer class](https://pytorch.org/rl/reference/trainers.html) doesn't support multiple loss modules or optimisers.

## Solution

I wish there will be an example code of how to implement the aforementioned algorithms, or alternatively, good guidance on how to customise the Trainer.

## Checklist

- [ x] I have checked that there is no similar issue in the repo

**Comment by vmoens** [comment]
Hi @levilovearch 
thanks for the issue!
I think in this case it's more a problem of doc than features.
We have an [`OptimizerHook`](https://github.com/pytorch/rl/blob/e79f15ef89b0257ae9203fe82788d092b386402a/torchrl/trainers/trainers.py#LL692C38-L692) to which you'd give the entry of a loss dict and it will backprop through that and optimizer the relevant weights.
Is that what you have in mind, or is there something more that is needed?

**Comment by levilovearch** [comment]
Hi @vmoens 

Thanks for your help. After carefully reading the doc, I tend to agree with you that It's more like a doc problem.
